engine/gamesys/src/gamesys/scripts/script_particlefx.cpp

```diff
@@ -106,13 +106,15 @@ namespace dmGameSystem
 
             EmitterStateChangedCallbackArgs args(data.m_ComponentId, emitter_id, emitter_state);
 
-            if (!dmScript::InvokeCallback(data.m_CallbackInfo, EmitterStateChangedCallbackArgsCB, &args))
+            int ok = dmScript::InvokeCallback(data.m_CallbackInfo, EmitterStateChangedCallbackArgsCB, &args);
+            if (!ok)
             {
-                dmLogError("Could not run particlefx callback because the instance has been deleted.");
+                dmLogWarning("Could not run particlefx callback because the instance has been deleted.");
             }
 
-            // The last emitter belonging to this particlefx har gone to sleep, release lua reference.
-            if(num_awake_emitters == 0 && emitter_state == dmParticle::EMITTER_STATE_SLEEPING)
+            // Release the lua ref to the callback if we failed to invoke callback or if the last emitter
+            // belonging to this particlefx har gone to sleep.
+            if(!ok || (num_awake_emitters == 0 && emitter_state == dmParticle::EMITTER_STATE_SLEEPING))
```

Comment:
Might as well destroy the callback if we failed to invoke it. Is there any chance it will be successful the next time?

Reply:
I think that's a good approach, yes.